app/models/file_depot_s3.rb

```diff
@@ -0,0 +1,95 @@
+class FileDepotS3 < FileDepot
+  attr_accessor :s3
+
+  def self.uri_prefix
+    "s3"
+  end
+
+  def self.validate_settings(settings)
+    _log.debug("Validating settings [#{settings}]")
+    new(:uri => settings[:uri]).verify_credentials(nil, settings.slice(:username, :password))
+  end
+
+  extend ActiveSupport::Concern
+
+  #
+  # Connections
+  #
+  def connect(options = {})
+    username = options[:username] || authentication_userid(options[:auth_type])
+    password = options[:password] || authentication_password(options[:auth_type])
+    service  = :S3
+    # Note: The hard-coded aws_region will be removed after manageiq-ui-class implements region selection
+    aws_region = options[:region] || "us-east-1"
+
+    raw_connect(username, password, service, aws_region)
+  end
+
+  def with_depot_connection(options = {})
+    raise _("no block given") unless block_given?
+    _log.info("Connecting through #{self.class.name}: [#{name}]")
+    yield connect(options)
+  end
+
+  def verify_credentials(auth_type = nil, options = {})
+    _log.debug("verifying credentials for auth_type [#{auth_type}] with options [#{options}]")
+
+    connection_rescue_block do
+      # EC2 does Lazy Connections, so call a cheap function
+      with_depot_connection(options.merge(:auth_type => auth_type)) do |ec2|
+        validate_connection(ec2)
+      end
+    end
+
+    true
+  end
+
+  #
+  # Connections
+  #
+  def raw_connect(access_key_id, secret_access_key, service, aws_region)
```

Comment:
It looks like this is only being used once, can we just move this content into the `#connect` method?